### ISSUE TITLE
[BUG] Video files visibly frozen when used inside Browsersource on macOS

### ISSUE THREAD

**Issue by PatTheMav** [obsproject/obs-browser#213]
## Platform

Operating system and version: macOS 10.15.4
OBS Studio version: 24.0.6

## Expected Behavior
Video files contained in HTML pages inside browser sources should play back visibly.

## Current Behavior
Only audio part of video files is played back, the video is frozen on first frame.

## Steps to Reproduce
1. Create local HTML file with embedded video:

```
<html>
<body>
<video autoplay controls>
<source src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4" type="video/mp4">
</video>
</body>
</html>
``` 
2. Add browser source to a scene
3. Setup local HTML file in browser source
4. Observe video being stuck on first frame
5. Hear the music/audio part of the video played back normally

## Additional information
* When debugging the browser source using the `remote-debugging-port` the video can be seen playing in the debugger's preview window.
* The video does not visibly play in recordings either

**Comment by vexir** [comment]
Discovered this when trying to use an animated overlay from StreamElements. Am stuck with locally driven or static overlays until a fix. Thanks @PatTheMav for the help!

**Comment by pkviet** [comment]
can you test on previous releases ? v24 and v23 ? so that we spot when regression occured. Thanks

**Comment by vexir** [comment]
> can you test on previous releases ? v24 and v23 ? so that we spot when regression occured. Thanks

Happy to test. What subversions of either? Just 24.0.0 and 23.0.0?

**Comment by pkviet** [comment]
last subversions will do

**Comment by vexir** [comment]
> last subversions will do

@pkviet Since this occured on the latest version of Mac (24.0.6) I'll try 24.0.0 and 23.2.1? Will the latter work on Catalina?

**Comment by pkviet** [comment]
oh sorry, yeah try these

**Comment by PatTheMav** [comment]
Here are the results of a quick check:

* v23.2.1 - Video plays back as expected
* v24.0.2 - Video is frozen

**Comment by pkviet** [comment]
ah f... So it's a regression. Thanks

**Comment by vexir** [comment]
@pkviet any dangers in using 23.2.1 for now (on Catalina)?

**Comment by PatTheMav** [comment]
@pkviet Yeah I'm currently trying to build some other tags between 24.0.2 and 23.2.1 on macOS and check those as well.

**Comment by pkviet** [comment]
@PatTheMav no need; i know the main change. We upgraded cef betwwen v23 and v24. But I was pretty sure video was working. I'll have to confirm.
Does https://html5test.com/ display video support for h264 ?

**Comment by PatTheMav** [comment]
@pkviet Got it, would need to find and punch in the macOS 10.14 platform SDK anyway as otherwise the code doesn't compile on Catalina.

The test results are not readable at all inside the browser source, but I copied the results from the HTML:

```
video element Yes
Subtitles Yes
Audio track selection No
Video track selection No
Poster images Yes
Codec detection Yes

Video codecs
------------
MPEG-4 ASP support No
H.264 support Yes
H.265 support No
Ogg Theora support Yes
WebM with VP8 support Yes
WebM with VP9 support Yes
```

**Comment by vexir** [comment]
> @PatTheMav no need; i know the main change. We upgraded cef betwwen v23 and v24. But I was pretty sure video was working. I'll have to confirm.
> Does https://html5test.com/ display video support for h264 ?

How likely is it that it’s a CEF bug vs an OBS implementation bug?

**Comment by pkviet** [comment]
99% cef bug

**Comment by vexir** [comment]
Sad. So a fix is dependent on a CEF fix? Is it tracked in their issues as well? Or is this more of a rollback situation

**Comment by vexir** [comment]
@PatTheMav @pkviet I'm not sure if this is the same bug or just related, but I can't play local .webm files either. Doesn't work through the drag/drop method, doesn't work as a media source.

**Comment by pkviet** [comment]
I can play webm fine  ... So unrelated I'd say.

**Comment by vexir** [comment]
That was an unexpected answer, but saves me from trying to find another alpha supported video channel to export my overlay files to. Assuming I can figure out why it doesn't work. Sent you an email, @pkviet !

**Comment by dkonopka** [comment]
I have similar issues on MacOS - adding for example http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4 or http://dl5.webmfiles.org/big-buck-bunny_trailer.webm as browser source makes them freezed.

OBS: `25.0.7`
MacOS Catalina: `10.15.3`

It's iMac Late 2015:
- Graphics: `AMD Radeon R9 M380 2 GB`
- Processor:  3,2 GHz Quad-Core Intel Core i5`

**Comment by nathanlejeune** [comment]
Hey everyone!

A lot this over my head and I am learning as I go, but I have a brand new mac, Catalina 10.15.4 and I can not play Webm files through Streamelements. They just show a black shape the shape of the webm file but nothing is there. The webm file does play if you drag and drop but not if you use browser source. It also messes up the browser source on Streamelements which has to be resaved before it works.

**Comment by pkviet** [comment]
@nathanlejeune for issues with streamelements, check their support. They replace our browser source with their own ...

**Comment by julijane** [comment]
I would like to add to this issue that it happens on Linux as well. Not exactly the same, the video does not full stop (e.g. when playing youtube), but it is very choppy. On the other hand, the third-party obs-linuxbrowser plugin (which is now considered a dead project though) does not have the issue. obs-linuxbrowser uses CEF 3440, which appears to be the version used by this project before as well, so this makes it likely that the issue is really with the use of a newer (but also already quite old) CEF version.

**Comment by camalot** [comment]
I experience this in OBS 26.0.2 on Windows. 

The audio plays but video stuck on first frame. tried webm and mp4. 

I have not tried prior version of OBS yet.

**Comment by MisterCommand** [comment]
Hello,
I am experiencing something similar in OBS `26.0.2`, I tried several previous versions and it is finally working in OBS `23.2.1`. I tried launching YouTube embed video, and when the progress bar is hidden, frames stated to be skipped and became frozen. It returned to normal once I got into interaction mode and hovered above to show the progress bar. I also tested Twitch, normal Twitch worked fine but Twitch embed worked liked YouTube embed. Hope this helps.

My Machine:
i7-8750H
GTX 1070 MaxQ
Latest version of Windows and graphics drivers
Clean install of OBS

My PC's OBS works perfectly fine under the same scenario.

**Comment by PatTheMav** [comment]
As was mentioned this is a known issue with CEF 3770 that we ship with in 26.0.2. An update is in the works, but there are other issues we'd like to hammer out first before updating.

**Comment by WizardCM** [comment]
Thanks to the following PRs and the work of @PatTheMav and @pkviet - hardware acceleration is now available in OBS starting in [26.1.2](https://github.com/obsproject/obs-studio/releases/tag/26.1.2). This fixes the video playback issue mentioned.

* #252
* https://github.com/obsproject/obs-studio/pull/3933
* https://github.com/obsproject/obs-studio/pull/3934
* https://github.com/obsproject/obs-studio/pull/4034
